documentation/modules/oauth/con-mapping-keycloak-authz-services-to-kafka-model.adoc

```diff
@@ -0,0 +1,103 @@
+[id="con-mapping-keycloak-authz-services-to-kafka-model_{context}"]
+= Map Keycloak Authorization Services to the Kafka authorization model
+
+[role="_abstract"]
+
+The Kafka authorization model is used as a basis for defining the Keycloak roles and resources that will control access to Kafka.
+
+To grant Kafka permissions to user accounts or service accounts, you first create an _OAuth client specification_ in Keycloak for the Kafka broker.
+You then specify Keycloak Authorization Services rules on the client.
+Typically, the client id of the OAuth client that represents the broker is `kafka` (the xref:proc-oauth-authorization-keycloak-example_-{context}[example files] provided with Strimzi use `kafka` as the OAuth client id).
+
+[NOTE]
+====
+If you have multiple Kafka clusters, you can use a single OAuth client (`kafka`) for all of them.
+This gives you a single, unified space in which to define and manage authorization rules.
+However, you can also use different OAuth client ids (for example, `my-cluster-kafka` or `cluster-dev-kafka`) and define authorization rules for each cluster within each client configuration.
+====
+
+The `kafka` client definition must have the *Authorization Enabled* option enabled in the Keycloak Admin Console.
+
+All permissions exist within the scope of the `kafka` client. If you have different Kafka clusters configured with different OAuth client IDs, they each need a separate set of permissions even though they're part of the same Keycloak realm.
+
+When the Kafka client uses OAUTHBEARER authentication, the Keycloak authorizer (`KeycloakRBACAuthorizer`) uses the access token of the current session to retrieve a list of grants from the Keycloak server.
+To retrieve the grants, the authorizer evaluates the Keycloak Authorization Services policies and permissions.
+
+.Authorization scopes for Kafka permissions
+
+An initial Keycloak configuration usually involves uploading authorization scopes to create a list of all possible actions that can be performed on each Kafka resource type.
+This step is performed once only, before defining any permissions.
+You can add authorization scopes manually instead of uploading them.
+
+Authorization scopes must contain all the possible Kafka permissions regardless of the resource type:
```

Comment:
Just to clarify, using 'should' instead of 'must' was also good. Scopes that are not present can't be targeted, that's all. For example if 'IdempotentWrite' is not present, you can't create a rule matching IdempotentWrite action on a resource (e.g. some topic). By saying 'must' we just maybe reduce potential issues later when uses can't seem to target some permission because it's not listed under authorization scope. Thus, not listing all scopes will not result in any immediate errors. If you know what you're doing, for example that you'll never need IdempotentWrite, then you can as well omit it. But it's not bad to say 'must' either as I said - to minimise the need to fix authorization scopes list later on.

Reply:
Very helpful, thanks. I added a brief note:

> If you're certain you won't need a permission (for example, `IdempotentWrite`), you can omit it from the list of authorization scopes. However, that permission won't be available to target on Kafka resources.